extensions/amp-a4a/0.1/amp-a4a.js

```diff
@@ -615,31 +599,45 @@ export class AmpA4A extends AMP.BaseElement {
           //    render-in-DOM failed, and no ad would be displayed.  However,
           //    all of the enclosed mutations are fairly simple and unlikely
           //    to fail.
-          this.vsync_.mutate(() => {
-            const doc = this.element.ownerDocument;
-            // Copy fonts to host document head.
-            this.relocateFonts_(creativeMetaData);
-            // Create and setup shadow root.
-            const shadowRoot = createShadowEmbedRoot(this.element,
-                creativeMetaData.customElementExtensions || []);
-            // Add custom CSS.
-            const customStyle = doc.createElement('style');
-            customStyle.setAttribute('amp-custom', '');
-            customStyle.textContent = cssBlock;
-            shadowRoot.appendChild(customStyle);
-            // Add body.
-            const bodyAttrString = creativeMetaData.bodyAttributes ?
-                  ' ' + creativeMetaData.bodyAttributes : '';
-            const temp = doc.createElement('div');
-            temp./*OK*/innerHTML =
-                `<${AMP_BODY_STRING}${bodyAttrString}></${AMP_BODY_STRING}>`;
-            const bodyElement = temp.firstElementChild;
-            shadowRoot.appendChild(bodyElement);
-            bodyElement./*OK*/innerHTML = bodyBlock;
-            this.rendered_ = true;
-            this.onAmpCreativeShadowDomRender();
+          // Create and setup friendly iframe.
+          const iframe = createElementWithAttributes(
+            this.element.ownerDocument, 'iframe', {
+              'frameborder': '0', 'allowfullscreen': '',
+              'allowtransparency': '', 'scrolling': 'no'});
+          this.applyFillContent(iframe);
+
+          // HTML is all but the extensions which must be removed from the
+          // creative.
+          // TODO(keithwrightbos): remove this monstrousity in favor of
+          // validation providing offsets to extension & AMP runtime
+          // locations.
+          const extensions =
+            creativeMetaData.customElementExtensions || [];
+          // Include v0.js for removal.
+          extensions.push('v0\.js');
+          extensions.forEach(extension => {
+            creative = creative.replace(new RegExp(
+              `<script[^>]+${extensionName}[^>]+>\s*</script\s*>\n?`),
```

Comment:
4 spaces here as well.

the previous line is fine though, as it's not a forced line break.